streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -635,18 +626,14 @@ private void to(final TopicNameExtractor<K, V> topicExtractor,
     @Override
     public KTable<K, V> toTable(final Named named,
                                 final Materialized<K, V, KeyValueStore<Bytes, byte[]>> materialized) {
+        Objects.requireNonNull(named, "named can't be null");
         Objects.requireNonNull(materialized, "materialized can't be null");
 
-        final MaterializedInternal<K, V, KeyValueStore<Bytes, byte[]>> materializedInternal =
-            new MaterializedInternal<>(materialized);
```

Comment:
Creating this `MaterializedInternal` was actually a bug as if the passed in "store name" was `null`, no store name would be generated, and we would fail with an NPE. It's fixed implicitly with the refactoring (this bug was exposed after I rewrote a test -- leave a comment there, too)